### PR TITLE
Sawfish (git): New Package

### PR DESCRIPTION
**Pull request by AndersonTorres**
Sawfish is a versatile, Lisp-based window manager

In that commit I include all Sawfish stack:
- librep, a lisp system;
- rep-gtk, bindings for gtk
- sawfish, the window manager

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -10902,6 +10902,10 @@ let
     fontsConf = makeFontsConf { fontDirectories = [ freefont_ttf ]; };
   };
 
+  librep = callPackage ../development/libraries/librep/git.nix { };
```
Comment (jagajaga) [inline pkgs/top-level/all-packages.nix]: Maybe add \n between lines?

File: pkgs/development/libraries/librep/git.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+, readline, texinfo
+, gdbm, gmp, libffi }:
+
+let
+  rev = "a1f2db721aa5055e90f6a76fde625946340ed8cf";
+in
+with stdenv.lib;
+stdenv.mkDerivation rec {
+
+  name = "librep-${rev}";
```
Comment (jagajaga) [inline pkgs/development/libraries/librep/git.nix]: Don't make revision as a name part. 
Use date of that commit instead. Please use librep-yyyy-mm-dd.
Apply this to all PR, please.

**Comment by jagajaga** [comment]
And, IMO, don't use git.nix. Rename it to default.nix.

**Comment by jagajaga** [comment]
You can use default.nix, but change names to `name-git-yyyy-mm-dd`.

File: pkgs/development/libraries/rep-gtk/git.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{ stdenv, fetchgit, pkgconfig, autoreconfHook, librep, gtk2 }:
+
+let
+  rev = "74ac3504f2bbbcc9ded005ab97cbf94cdc47924d";
+in
+with stdenv.lib;
+stdenv.mkDerivation rec {
+
+  name = "rep-gtk-${rev}";
```
Comment (jagajaga) [inline pkgs/development/libraries/rep-gtk/git.nix]: So date here.

File: pkgs/applications/window-managers/sawfish/git.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
+{ stdenv, fetchgit, pkgconfig, which, autoreconfHook, rep-gtk, pango, gdk_pixbuf, libXinerama, libXrandr, libXtst, imlib, gettext, texinfo, makeWrapper }:
+
+let
+  rev = "44729f44017e6779b4b66a7ecdbd63a98731f668";
+in
+with stdenv.lib;
+stdenv.mkDerivation rec {
+
+  name = "sawfish-${rev}";
```
Comment (jagajaga) [inline pkgs/applications/window-managers/sawfish/git.nix]: And here :)

**Comment by jagajaga** [comment]
But anyway PR is good :+1:

**Comment by AndersonTorres** [comment]
Done!

**Comment by jagajaga** [comment]
For every pkgs you submitted there is a release. Why not use them? Or explain why do you use HEAD.

**Comment by AndersonTorres** [comment]
Oh, yes, of course. I am using the git version because the "numbered release" needs a whole bunch of annoying patches. Well, the git version needs too, but it is a lot easier to work with sed commands.
Also, the numbered release is written like a monolithic piece: librep, rep-gtk and sawfish need to be installed in a same filesystem tree, something Nix doesn't like.

I even contact the Sawfish mailing list, and they were very receptive to my message! They explained the git version is more suitable to Nix (in fact they have cited Guix), and I tested it.

**Comment by jagajaga** [comment]
That's nice! So git version seems to be what we need.

But anyway, can you change name to `name-git-YYYY-MM-DD`? :)

**Comment by AndersonTorres** [comment]
Done!

**Comment by jagajaga** [comment]
Merged in https://github.com/NixOS/nixpkgs/commit/59d2ab656350991e6f8010a17a6ec72b2349984c
Check names there - I've changed them. That's what I wanted :)
Thank you for this work!

File: pkgs/applications/window-managers/sawfish/default.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
+{ stdenv, fetchgit, pkgconfig, which, autoreconfHook, rep-gtk, pango, gdk_pixbuf, libXinerama, libXrandr, libXtst, imlib, gettext, texinfo, makeWrapper }:
+
+with stdenv.lib;
+stdenv.mkDerivation rec {
+
+  name = "sawfish-git-20150215";
```
Comment (jagajaga) [inline pkgs/applications/window-managers/sawfish/default.nix]: Hmm, anyway - you did wrong a bit, the date of commit is 31 Jan (not 15 Feb) as I can see in their repo.
Let's forgive it now, but hope this won't happen again.

**Pull request closed without merging**